tests/backend/test_beamsplitter_operation.py

```diff
@@ -151,3 +153,55 @@ def test_coherent_vacuum_interfered_fock_elements(
             )
 
         assert np.allclose(numer_state, ref_state, atol=tol, rtol=0)
+
+
+@pytest.mark.backends("fock")
+class TestModeSubsets:
+    """Tests on mode subsets using the Fock basis representation."""
+
+    @pytest.mark.parametrize("t", T_VALUES)
+    @pytest.mark.parametrize("mag_alpha", MAG_ALPHAS[1:])
+    @pytest.mark.parametrize("r_phi", PHASE_R)
+    def test_beamsplitter_on_mode_subset(
+            self, setup_backend, mag_alpha, t, r_phi, cutoff, pure, tol
+    ):
+        """Tests that apply the BS on different mode subsets."""
+
+        phase_alpha = np.pi / 5
+        alpha = mag_alpha * np.exp(1j * phase_alpha)
+        r = np.exp(1j * r_phi) * np.sqrt(1.0 - np.abs(t) ** 2)
+
+        n_modes = 4
+        for modes in it.combinations(range(n_modes), 2):
```

Comment:
This for loop can be moved into a `@pytest.mark.parametrize()` decorator. The advantage of doing this is that pytest will treat each parametrized loop iteration as a separate test, and failed tracebacks will include the mode list it failed on.